src/AggregateFunctions/AggregateFunctionQuantile.cpp

```diff
@@ -27,7 +27,7 @@ AggregateFunctionPtr createAggregateFunctionQuantile(
     const std::string & name, const DataTypes & argument_types, const Array & params, const Settings *)
 {
     /// Second argument type check doesn't depend on the type of the first one.
-    Function<void, true>::assertSecondArg(argument_types);
+    Function<UInt8, true>::assertSecondArg(argument_types);
```

Comment:
After making `hasTrivialDestructor()` constexpr, instantiating the template with void was no longer valid, so I changed it to `UInt8` instead.

Reply:
I don't understand how `hasTrivialDestructor()` affects it.

Reply:
This happened because in order to validate `std::is_trivially_destructible_v` at compile time, the compiler is instantiating the template with the type (which used to be `void`) but there are a few things in those classes that can't work when `T=void`:

```
/mnt/ch/ClickHouse/src/AggregateFunctions/ReservoirSampler.h:82:25: error: cannot form a reference to 'void'
    void insert(const T & v)
                        ^
/mnt/ch/ClickHouse/src/AggregateFunctions/QuantileReservoirSampler.h:27:10: note: in instantiation of template class 'ReservoirSampler<void, ReservoirSamplerOnEmpty::RETURN_NAN_OR_ZERO>' requested here
    Data data;
         ^
/mnt/ch/ClickHouse/contrib/libcxx/include/__type_traits/is_trivially_destructible.h:24:38: note: in instantiation of template class 'DB::QuantileReservoirSampler<void>' requested here
    : public integral_constant<bool, __is_trivially_destructible(_Tp)> {};
                                     ^
/mnt/ch/ClickHouse/contrib/libcxx/include/__type_traits/is_trivially_destructible.h:47:53: note: in instantiation of template class 'std::is_trivially_destructible<DB::QuantileReservoirSampler<void>>' requested here
inline constexpr bool is_trivially_destructible_v = is_trivially_destructible<_Tp>::value;
                                                    ^
/mnt/ch/ClickHouse/src/AggregateFunctions/IAggregateFunction.h:706:72: note: in instantiation of variable template specialization 'std::is_trivially_destructible_v<DB::QuantileReservoirSampler<void>>' requested here
    constexpr bool hasTrivialDestructor() const override { return std::is_trivially_destructible_v<Data>; }
```

```
/mnt/ch/ClickHouse/src/Common/PODArray_fwd.h:33:63: error: invalid application of 'sizeof' to an incomplete type 'void'
          size_t rounded_bytes = integerRoundUp(inline_bytes, sizeof(T))>
                                                              ^~~~~~~~~
/mnt/ch/ClickHouse/src/AggregateFunctions/ReservoirSampler.h:234:23: note: in instantiation of default argument for 'PODArrayWithStackMemory<void, 64UL>' required here
    using Array = DB::PODArrayWithStackMemory<T, 64>;
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/ch/ClickHouse/src/AggregateFunctions/QuantileReservoirSampler.h:27:10: note: in instantiation of template class 'ReservoirSampler<void, ReservoirSamplerOnEmpty::RETURN_NAN_OR_ZERO>' requested here
    Data data;
         ^
/mnt/ch/ClickHouse/contrib/libcxx/include/__type_traits/is_trivially_destructible.h:24:38: note: in instantiation of template class 'DB::QuantileReservoirSampler<void>' requested here
    : public integral_constant<bool, __is_trivially_destructible(_Tp)> {};
```